urllib3/connection.py

```diff
@@ -162,6 +162,31 @@ def connect(self):
         conn = self._new_conn()
         self._prepare_conn(conn)
 
+    def request_chunked(self, method, url, body=None, headers={}):
+        """
+        Alternative to the common request method, which sends the
+        body with chunked encoding and not as one block
+        """
+        header_names = set(k.lower() for k in headers)
+        self.putrequest(method, url, skip_accept_encoding=True)
+        for header, value in headers.items():
+            self.putheader(header, value)
+        if 'transfer-encoding' not in header_names:
+            self.putheader('Transfer-Encoding', 'chunked')
+        self.endheaders()
+
+        if body is not None:
+            for chunk in body:
+                if not chunk:
+                    continue
+                self.send(hex(len(chunk))[2:].encode('utf-8'))
```

Comment:
As a follow-up, it is also subtly wrong.

There is an unconditional encode of the chunk to UTF-8 two lines below. If a Unicode string is passed, the `len()` call here may return a smaller number of bytes than the actual number used in the chunk. Given that the unconditional call to `.encode` doesn't work on Python 3 for byte string chunks, I recommend that this method simply not accept Unicode strings at all: remove the call to `.encode` from the chunk send and it essentially becomes mandatory to pass this code bytes.